[DealabsBridge] Add new bridge

sysadminstory opened the pull request:
This is a new Bridge for the Deal search function of dealabs.com

The html content is a little bit crappy, but it works !

sysadminstory committed:
[DealabsBridge] Add new bridge

logmanoriginal commented on bridges/DealabsBridge.php:
```diff
@@ -0,0 +1,134 @@
+	const PARAMETERS = array( array (
+		'q' => array('name' => 'Mot(s) clé(s)', 'type' => 'text', 'required' => true ),
+		'expired_choice' => array('name' => 'Afficher deals expirés', 'type' => 'checkbox'),
+		'instore_choice' => array('name' => 'Afficher deals en magasin', 'type' => 'checkbox'),
```
This part looks a bit off. Please indent it like the arrays below (one element per line)

logmanoriginal commented on bridges/DealabsBridge.php:
```diff
@@ -0,0 +1,134 @@
+			or returnServerError('Could not request Dealabs.');
+		$list = $html->find('article');
+		if($list === null) {
+			return;
```
Please return some kind of error message. Maybe something like this:

`returnClientError('Your combination of parameters returned no results');`

It is important for the user to receive some kind of feedback if something goes wrong.

logmanoriginal commented on bridges/DealabsBridge.php:
```diff
@@ -0,0 +1,134 @@
+			$item['uri'] = $deal->find('a.title',0)->href;
+			$item['title'] = $deal->find('a.title', 0)->plaintext;
+			$item['author'] = $deal->find('a.poster_link', 0)->plaintext;
+			$item['content'] = '<table><tr><td>' .$deal->find('div.image_part',0)->outertext . '</td><td>'. $deal->find('a.title',0)->outertext . $deal->find('p.description',0)->outertext .'</td><td>'. $deal->find('div.vote_part',0)->outertext . '</td></table>';
```
Please split this line into multiple ones in order to shrink the length of each single line to less than 80 characters. You can use the dot as first character for each line to make it easier to read:

```PHP
$item['content'] = '<table><tr><td>' 
. $deal->find('div.image_part',0)->outertext 
. '</td><td>'
. $deal->find('a.title',0)->outertext 
. $deal->find('p.description',0)->outertext 
. '</td><td>'
. $deal->find('div.vote_part',0)->outertext 
. '</td></table>';
```

logmanoriginal commented on bridges/DealabsBridge.php:
```diff
@@ -0,0 +1,134 @@
+
+	private function relativeDateToTimestamp($str) {
+		$date = new DateTime();
+		$date->modify(str_replace(array('il y a ', 'min' , 'h', 'jour', 'jours', 'mois', 'ans'), array('-', 'minute', 'hour', 'day', 'month', 'year'), $str));
```
Same as above, this line is a bit too long.

logmanoriginal reviewed (changes requested):
Nice job making this bridge with so many options to choose from 👍 

I added some comments regarding error handling and line length (see below).

For some reason Travis is broken right now, so it always fails without performing any tests. 

There are definitely some spaces missing around commas in your code, so we'll have to sit tight and cross our fingers that Travis gets better soon.

sysadminstory committed:
[DealabsBridge] Conform to coding policy

- If no there are no results, an explicit message is now returned
- Commas are now following the coding policy
- Lines are no longer more than 80 chars when possible

sysadminstory commented:
Getting all the categories together was not the toughest in this Bridge, the HTML was a little bit harder  :sunglasses:

I followed your advices and updated the code.

There actually one line longer than 80 chars, but splitting this line would IMHO make it less readable.

I hope I did not forgot anything !

logmanoriginal commented:
Thanks for updating.

>There actually one line longer than 80 chars, but splitting this line would IMHO make it less readable.

That's fine. 80 characters are a rule of thumb. The checks will definitely fail if a line exceeds 120 characters.

I did an offline check on your last commit and there are still some errors:

```
FILE: /var/www/html/rss-bridge/bridges/DealabsBridge.php
----------------------------------------------------------------------
FOUND 6 ERRORS AFFECTING 6 LINES
----------------------------------------------------------------------
   6 | ERROR | [ ] String concat is not required here; use a single
     |       |     string instead
     |       |     (Generic.Strings.UnnecessaryStringConcat.Found)
   7 | ERROR | [ ] String concat is not required here; use a single
     |       |     string instead
     |       |     (Generic.Strings.UnnecessaryStringConcat.Found)
  89 | ERROR | [x] Whitespace found at end of line
     |       |     (Squiz.WhiteSpace.SuperfluousWhitespace.EndLine)
 157 | ERROR | [x] Expected 1 space after ">="; 0 found
     |       |     (Squiz.WhiteSpace.OperatorSpacing.NoSpaceAfter)
 160 | ERROR | [x] Expected 1 space after closing brace; newline
     |       |     found
     |       |     (Squiz.ControlStructures.ControlSignature.SpaceAfterCloseBrace)
 166 | ERROR | [x] Expected 1 space after closing brace; newline
     |       |     found
     |       |     (Squiz.ControlStructures.ControlSignature.SpaceAfterCloseBrace)
----------------------------------------------------------------------
PHPCBF CAN FIX THE 4 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------

Time: 944ms; Memory: 18Mb
```

Regarding lines 6 and 7 you can do this:

```PHP
const DESCRIPTION = 'Return the Dealabs search result using keywords,
 with/without expired deals, with/without shop deals and by category';
```

Notice that there is a space at the beginning of the second line. The output is the same.

***

Travis is fixed in master. Unfortunately this PR cannot pass builds unless you rebase on master.

sysadminstory committed:
Merge remote-tracking branch 'upstream/master'

sysadminstory committed:
Coding style fix

Fixed the bridge to follow the project coding style

sysadminstory committed:
Delete usefull whitespace

sysadminstory commented:
I'm pretty new to teamwork with git, therefore it took so long ...
TIL to rebase from a fork :satisfied:

Now it's all fixed, and checked !

teromene commented:
Merged, thanks for your work !
